cmp/badge/Badge.ts

```diff
@@ -37,9 +41,17 @@ export const [Badge, badge] = hoistCmp.withFactory<BadgeProps>({
             classes.push('xh-badge--compact');
         }
 
+        const divProps = merge(
+            {className: classNames(className, classes)},
+            {style: layoutProps},
```

Comment:
Seems maybe a little unfortunate to have to pass the style props here since `box` already handles that (and these are designated `BoxProps`).  Should these component be returning a `box` instead of a `div`?  By accepting `BoxProps`, I wonder if that's the implication anyway?

Also wonder if we should make a utility method `splitBoxProps`, which would essentially do what `splitLayoutProps` does but with the addition of `TestSupportProps`, which we would just pass down to the underlying `box` cmp in all of these cases rather than having to destructure the `testId` for every component to handle

Reply:
The underlying issue (in my mind, if I recall correctly) is that `Box` gets a default flex layout, which was a mistake we made very early on and never revisited.  I really want it to simply be a `div` with some syntax candy, but instead (due to flex defaults) `box` is actually equiv to `hbox`.

https://github.com/xh/hoist-react/blob/d8319d3f28ff1aefd975d56a3624e245b305239b/cmp/layout/Box.ts#L37

For this badge case, maybe that would be fine, or even desirable, but it would be another kind of change that could break the layout of existing badges that have multiple child comps. I don't know how many of those there are - maybe none.  

So I would be open to it, but we should probably add a note to the CL.